project/FrontEnd/collaborative_science_platform/lib/providers/workspace_provider.dart

```diff
@@ -0,0 +1,361 @@
+import 'dart:convert';
+import 'package:collaborative_science_platform/exceptions/workspace_exceptions.dart';
+import 'package:collaborative_science_platform/models/workspaces_page/workspace.dart';
+import 'package:collaborative_science_platform/models/workspaces_page/workspaces.dart';
+import 'package:collaborative_science_platform/utils/constants.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+class WorkspaceProvider with ChangeNotifier {
+  Workspaces? workspaces;
+  Workspace? workspace;
+
+  void clearAll() {
+    workspace = null;
+    workspaces = null;
+  }
+
+  Future<void> getUserWorkspaces(int id) async {
+    Uri url = Uri.parse("${Constants.apiUrl}/get_user_workspaces/?user_id=$id");
+    final Map<String, String> headers = {
+      "Accept": "application/json",
+      "content-type": "application/json"
```

Comment:
Shouldn't we add user token here? If not something wrong with backend. Workspace API should be protected with Auth (user token)